docs/client-specifications.rst

```diff
@@ -0,0 +1,246 @@
+
+Clients SHOULD NOT rely on arbitrary server side filtering. In Remote Settings, collections are quite small anyway, and can usually be fetched entirely to be filtered on the client side. This helps us reduce our CDN cache cardinality.
+
+
+**Poll for changes**:
+
+``GET /v1/buckets/monitor/collections/changes/changeset?_expected={timestamp}``.
+
+Returns the list of collections and their current timestamp.
+
+- ``changes``: list of collections and their timestamp, optionally filtered with ``?_since="{timestamp}"``
+- ``tiemstamp``: highest collections timestamp
+
+.. note::
+
+    The ``_expected={}`` querystring parameter is mandatory but can be set to ``0`` if unknown. See next section about cache busting.
+
+Examples:
+
+* `get_latest_change_timestamp() in remote-settings-client <https://github.com/mozilla-services/remote-settings-client/blob/2538d6a07c28a3966b996d52596807df8c37130d/src/client/kinto_http.rs#L79-L105>`
+* `fetchLatestChanges() in Gecko <https://searchfox.org/mozilla-central/rev/1f27a4022f9f1269d897526c1c892a57743e650c/services/settings/Utils.sys.mjs#376-457>`_
+
+
+Cache Busting
+'''''''''''''
+
+Using push notifications:
+
+.. image:: images/client-specifications-cache-bust.png
+
+.. https://mermaid-js.github.io/mermaid-live-editor/
+.. sequenceDiagram
+..     participant Remote Settings
+..     participant Push Server
+..     participant CDN
+..     participant Client
+..     Remote Settings->>Push Server: Publish [timestamp]
+..     Push Server->>Client: Broadcast [timestamp]
+..     Client->>+CDN: Poll changes [timestamp]
+..     CDN->>Remote Settings: Cache miss|hit [url]
+..     Remote Settings-->>CDN:
+..     CDN-->>-Client: Modified collections [Array[timestamp]]
+..     Client->>+CDN: Fetch collection changeset [timestamp]
+..     CDN->>Remote Settings: Cache miss|hit [url]
+..     Remote Settings-->>CDN:
+..     CDN-->>-Client: Changeset [data, metadata, timestamp]
+
+Without push notifications:
+
+.. image:: images/client-specifications-cache-poll.png
+
+.. https://mermaid-js.github.io/mermaid-live-editor/
+.. sequenceDiagram
+..     participant Remote Settings
+
+..     participant CDN
+..     participant Client
+
+..     Client->>+CDN: Poll changes [timestamp=0]
+..     CDN->>Remote Settings: TTL expired|hit [url]
+..     Remote Settings-->>CDN:
+..     CDN-->>-Client: Modified collections [Array[timestamp]]
+..     Client->>+CDN: Fetch collection changeset [timestamp]
+..     CDN->>Remote Settings: TTL expired|hit [url]
+..     Remote Settings-->>CDN:
+..     CDN-->>-Client: Changeset [data, metadata, timestamp]
+
+
+Environment Switching
+'''''''''''''''''''''
+
+Clients MAY offer a convenient way to switch before DEV, STAGE, or PROD environments, in order to facilitate the work of QA teams.
+
+Clients SHOULD use PROD by default. And for security reasons, they must be some protection in place to prevent users to switch environments.
```

Comment:
I think "they" should be "there". 
And "prevent users to switch" maybe "prevent users from switching".